src/pass/llvmgen.cpp

```diff
@@ -507,6 +572,54 @@ llvm::Value* LLVMGen::visit(Sym old_sym)
     return var;
 }
 
+void LLVMGen::init_cuda()
+{
+    InitializeAllTargets();
+    InitializeAllTargetMCs();
+    InitializeAllAsmPrinters();
+    InitializeAllAsmParsers();
+
+    llmod()->setTargetTriple("nvptx64-nvidia-cuda");
+}
+
+TargetMachine* LLVMGen::get_target()
+{
+    std::string Error;
+    const llvm::Target* Target =
+        llvm::TargetRegistry::lookupTarget("nvptx64-nvidia-cuda", Error);
+
+    llvm::TargetOptions opt;
+    llvm::TargetMachine* TM =
+        Target->createTargetMachine("nvptx64-nvidia-cuda", "sm_52",
+                                    "+ptx76",  // PTX version
+                                    opt, llvm::Reloc::Static);
+
+    llmod()->setDataLayout(TM->createDataLayout());
+
+    return TM;
+}
```

Comment:
remove these functions from llvmgen. The only purpose of llvmgen is to generate the llvmcode . it should do anything related to executing it. Execution related stuff belong to the engine